packages/grafana-runtime/src/services/SidecarService_EXPERIMENTAL.ts

```diff
@@ -102,14 +134,31 @@ export class SidecarService_EXPERIMENTAL {
    * @experimental
    */
   openAppV2(pluginId: string, path?: string) {
-    if (!this.assertFeatureEnabled()) {
+    if (!(this.assertFeatureEnabled() && this.mainOnAllowedRoute)) {
       return;
     }
 
+    this.mainLocationWhenOpened = this.mainLocationService.getLocation().pathname.match(pathRegex)?.[0];
     this.memoryLocationService.push({ pathname: `/a/${pluginId}${path || ''}` });
     reportInteraction('sidecar_service_open_app', { pluginId, version: 2 });
   }
 
+  /**
+   * Opens an app in a sidecar. You can also relative path inside the app to open.
+   * @experimental
+   */
+  openAppV3(options: { pluginId: string; path?: string; follow?: boolean }) {
+    if (!(this.assertFeatureEnabled() && this.mainOnAllowedRoute)) {
+      return;
+    }
+
+    this.follow = options.follow || false;
```

Comment:
Not sure if `follow` should be `true` by default. Guess doesn't really matter at this point.

Reply:
Sorry you think it should be true by default or you think it is true by default and don't think it should be? 😅

Reply:
Haha, yea, unclear message.

I was wondering if it shouldn't be `true` by default, instead of defaulting to `false`. But it's a nit really and maybe `false` is also good since it's more "restrictive".